### PR TITLE
[cli] add `platform` command which returns current platform

### PR DESCRIPTION
**Pull request by kghost**
Sometime we need to identify which platform we are currently using, especially in a test cases, where a unified interface is provided to control devices, the test cases have no idea about current platform.

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -5463,6 +5463,13 @@ template <> otError Interpreter::Process<Cmd("ping")>(Arg aArgs[])
 }
 
 #endif // OPENTHREAD_CONFIG_PING_SENDER_ENABLE
+       //
```
Comment (abtink) [inline src/cli/cli.cpp:5466]: Remove the extra `//`?

**Review by abtink — APPROVED** [review]
LGTM. Thanks. 👍

File: src/cli/cli.cpp [hunk 2]
```diff
@@ -7322,6 +7329,7 @@ otError Interpreter::ProcessCommand(Arg aArgs[])
 #if OPENTHREAD_CONFIG_PING_SENDER_ENABLE
         CmdEntry("ping"),
 #endif
+        CmdEntry("platform"),
```
Comment (zhanglongxia) [inline src/cli/cli.cpp:7332]: Should we add command description to README.md?

File: src/cli/cli.cpp [hunk 3]
```diff
@@ -5464,6 +5464,13 @@ template <> otError Interpreter::Process<Cmd("ping")>(Arg aArgs[])
 
 #endif // OPENTHREAD_CONFIG_PING_SENDER_ENABLE
 
+template <> otError Interpreter::Process<Cmd("platform")>(Arg aArgs[])
+{
+    OT_UNUSED_VARIABLE(aArgs);
```
Comment (superwhd) [inline src/cli/cli.cpp:5469]: May be good to verify if aArgs is empty.

**Review by superwhd — APPROVED** [review]
LGTM with a nit

File: src/cli/cli.cpp (hunk 3 above)
Comment (kghost) [reply src/cli/cli.cpp:5469]: I saw lots of commands doesn't check it. Maybe here we don't want to enforce args being empty to leave room for future enhancement, such that we can add args in the future w/o breaking previous version.

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]

**Pull request merged**